pandas/core/algorithms.py

```diff
@@ -1534,40 +1535,73 @@ def _take_nd_object(arr, indexer, out, axis: int, fill_value, mask_info):
 }
 
 
+@functools.lru_cache(maxsize=128)
+def __get_take_nd_function_cached(ndim, arr_dtype, out_dtype, axis):
+    """
+    Part of _get_take_nd_function below that doesn't need the mask
```

Comment:
this coment does't make any sense w/o the PR context. can you put / move a doc-string here. typing a +1

Reply:
The "and thus can be cached" on the next line is the essential continuation of the sentence. 
The mask can be an array, and thus is not hashable and thus cannot be used as argument for a cached function.